lib/dpl/provider/pages.rb

```diff
@@ -188,6 +228,8 @@ def push_app
         print_step "Starting deployment of #{@target_branch} branch to GitHub Pages..."
         print_step "The deployment is configured to preserve the target branch if it exists on remote" if @keep_history
         Dir.mktmpdir do |tmpdir|
+            github_install_key(tmpdir)
```

Comment:
This probably should go to `needs_key` and be called only once, instead of doing this in two places.